### PR TITLE
state.sls hangs on file.recurse with clean: True on windows

### PR DESCRIPTION
**Pull request by ushmodin**
### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/13654

### Tests written?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Review by twangboy — CHANGES_REQUESTED** [review]
Please submit against 2017.7 branch as well.

File: salt/states/file.py [hunk 1]
```diff
@@ -627,7 +627,7 @@ def _clean_dir(root, keep, exclude_pat):
             while True:
                 fn_ = os.path.dirname(fn_)
                 real_keep.add(fn_)
-                if fn_ in ['/', ''.join([os.path.splitdrive(fn_)[0], '\\\\'])]:
+                if fn_ in ['/', ''.join([os.path.splitdrive(fn_)[0], '\\'])]:
```
Comment (twangboy) [inline salt/states/file.py]: Actually, in my testing, this is causing file.recurse to hang...
If `\\` fixes yours and `\\\\` fixes mine, then we need to do something like this:
```
                if fn_ in [
                        os.sep,
                        ''.join([os.path.splitdrive(fn_)[0], os.sep]),
                        ''.join([os.path.splitdrive(fn_)[0], os.sep, os.sep]),
                        ]:
```

**Review by twangboy — APPROVED** [review]

**Comment by ushmodin** [comment]
@twangboy 
sorry, what do you mean

>> submit against 2017.7 branch as well

create new pull request for branch 2017?

**Comment by twangboy** [comment]
@ushmodin 
Yeah. In fact, I would go back as far as 2016.11. Those changes will be merged forward. So if you submit your PR against 2016.11, the changes will be merged forward into 2017.7 and develop. If you submit against develop, the change won't be available until the next major release... Next year sometime...

You should just be able to rebase your branch against 2016.11 and then push it up to Github. Then in Github, change the branch to 2016.11 (clicking the edit button at the top of this PR gives you that option).

So, you can either rebase, or open a new PR against 2016.11... whichever one you find easier.

**Comment by cachedout** [comment]
@ushmodin Will you be able to change the base branch here? Let us know if you need any help. Thanks!

**Comment by ushmodin** [comment]
1. i tried to cherry-pick commits from develop to 2016.11, but it did not work out

2. I duplicated changes to the branch 2016.11 and create new PR  43151

**Pull request closed without merging**